persistent/Database/Persist/TH.hs

```diff
@@ -349,6 +349,8 @@ mEmbedded _ (FTTypeCon Just{} _) =
     Left Nothing
 mEmbedded ents (FTTypeCon Nothing (EntityNameHS -> name)) =
     maybe (Left Nothing) Right $ M.lookup name ents
+mEmbedded ents (FTTypePromoted (EntityNameHS -> name)) =
+    maybe (Left Nothing) Right $ M.lookup name ents
```

Comment:
Not sure about this. I basically just mirrored what is done for `FTTypeCon` when there is no module (ie the case above).

Reply:
Hmm - I'd say let's go with a `Left Nothing` on all cases here. If we hit this, it means we're trying to determine whether we need an embedded entity reference, which we shouldn't - the only place this could *possibly* be is if we've got an `FTApp` or similar, and we want the SQL type construction to proceed unchanged.

```suggestion
    Left Nothing
```

Like, we'll have `foo   (Proxy 'User)`. That'll get turned into `FTApp (FTypeCon Nothing "Proxy") (FTTypePromoted "User")`. This code recurses and then finds that the corresponding `EmbedEntityDef` is for a `User`, even though that's totally wrong.